src/build/app/mod.rs

```diff
@@ -675,9 +676,36 @@ impl<'b> App<'b> {
         for arg in args.into_iter() {
             self.args.push(arg.into());
         }
+        self.panic_on_missing_help();
         self
     }
 
+    #[cfg(not(debug_assertions))]
+    fn panic_on_missing_help(&self) {}
+
+    #[cfg(debug_assertions)]
+    fn panic_on_missing_help(&self) {
+        if self.is_set(AppSettings::HelpRequired) {
+            let args_missing_help: Vec<String> = self
+                .args
+                .args
+                .iter()
+                .filter(|arg| arg.help.is_none() && arg.long_help.is_none())
+                .map(|arg| String::from(arg.name))
+                .collect();
+
+            if args_missing_help.len() > 0 {
+                let mut abort_message = String::from(
+                    "You activated the setting AppSettings::HelpRequired in the clap crate, but then did not implement help messages for all your arguments:\n"
+                );
+                for arg_name in args_missing_help {
+                    abort_message += &format!("Argument with name {} has neither help nor long_help implemented. Implement at least one!\n", arg_name);
+                }
```

Comment:
Can replace the above whole block with:

```rust
let abort_message = format!("AppSettings::HelpRequired is enabled, but some of your arguments does not have either `help` or `long_help` set: {}", arg_name.join(", "));
```

We shouldn't be too descriptive.

Reply:
Agreed